jax/interpreters/xla.py

```diff
@@ -59,6 +59,32 @@
 XlaComputationBuilder = Any  # xla_bridge._JaxComputationBuilder
 XlaExecutable = Any # xla_extension.LocalExecutable
 
+
```

Comment:
I think that the idea of having the context of a stack of nested primitives is a very good one. I would then require that `kind` is always a `Primitive`, and I would perhaps call it `PrimitiveStack`. Pushing this even further, why not include the primitive params in the stack, or have it be a `EqnStack`.